src/Symfony/Component/Notifier/Bridge/Smsapi/Tests/SmsapiTransportTest.php

```diff
@@ -44,4 +46,67 @@ public function unsupportedMessagesProvider(): iterable
         yield [new ChatMessage('Hello!')];
         yield [$this->createMock(MessageInterface::class)];
     }
+
+    public function createClient(int $statusCode, array $content): HttpClientInterface
+    {
+        $response = $this->createMock(ResponseInterface::class);
+        $response
+            ->method('toArray')
+            ->willReturn($content);
+
+        $response
+            ->method('getStatusCode')
+            ->willReturn($statusCode);
+
+        $client = $this->createMock(HttpClientInterface::class);
+        $client
+            ->method('request')
+            ->willReturn($response);
+
+        return $client;
+    }
+
+    public function responseProvider(): iterable
+    {
+        $responses = [
+            ['status' => 200, 'content' => '{"error":101,"message":"Authorization failed"}'],
+            ['status' => 500, 'content' => '{}'],
+            ['status' => 500, 'content' => '{"error":null,"message":"Unknown"}'],
+            ['status' => 500, 'content' => '{"error":null,"message":null}'],
+        ];
+
+        foreach ($responses as $response) {
+            yield [$response['status'], json_decode($response['content'], true)];
+        }
+    }
+
+    /**
+     * @dataProvider responseProvider
+     */
+    public function testThrowExceptionWhenMessageWasNotSent(int $statusCode, array $content)
+    {
+        $client = $this->createClient($statusCode, $content);
+        $transport = $this->createTransport($client);
+        $message = new SmsMessage('0611223344', 'Hello!');
+
+        $this->expectException(TransportException::class);
```

Comment:
If I don't miss anything, you can remove the second test and add the following to the first test:

```suggestion
        $this->expectException(TransportException::class);
        $this->expectExceptionMessage(HERE THE EXPECTED MESSAGE, add it to the provider);
```